src/lib/data/github-repo.ts

```diff
@@ -1,39 +1,93 @@
+import { GitHubRepository } from '../types';
+
 export interface GitHubApiResponse {
-  data?: any;
+  data?: { repository: GitHubRepository };
   error?: string;
 }
 
 export class GitHubRepo {
-  private readonly baseUrl: string;
+  private readonly graphqlUrl = 'https://api.github.com/graphql';
 
-  constructor(readonly owner: string, readonly repo: string) {
-    this.baseUrl = `https://api.github.com/repos/${owner}/${repo}`;
-  }
+  constructor(readonly owner: string, readonly repo: string) {}
 
   async metadata(): Promise<GitHubApiResponse> {
-    return this.fetchWithErrorHandling(this.baseUrl);
-  }
+    const query = `
+      query GetRepositoryEssentials($owner: String!, $name: String!) {
+        repository(owner: $owner, name: $name) {
+          stargazerCount
+          
+          # Root directory contents (for checking docs folders)
+          rootContents: object(expression: "HEAD:") {
+            ... on Tree {
+              entries {
+                name
+                type
+              }
+            }
+          }
+          
+          # README content variations (for checking examples)
```

Comment:
i thought we talked about readme being any file that includes the word `readme`? i think the right thing to do here (and then you tell me if its possible) is to:

a) grab the file names at the root
b) determine if there is a valid readme file 
c) query for the contents of that specific readme file